app/src/main/java/de/danoeh/antennapod/fragment/preferences/NetworkPreferencesFragment.java

```diff
@@ -174,4 +152,145 @@ private void showUpdateIntervalTimePreferencesDialog() {
+        int checkedItem = -1;
+        if(currInterval > 0)
+        {
+            String currIntervalStr = String.valueOf(TimeUnit.MILLISECONDS.toHours(currInterval));
+            checkedItem = ArrayUtils.indexOf(values, currIntervalStr);
+        }
+        ArrayAdapter<String> spinnerArrayAdapter = new ArrayAdapter<String>(getContext(), android.R.layout.simple_spinner_item, entries); //selected item will look like a spinner set from XML
+        spinnerArrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+        //TIMER
+        int hourOfDay = 7;
+        int minute = 0;
+        int[] updateTime = UserPreferences.getUpdateTimeOfDay();
+        if (updateTime.length == 2) {
+            hourOfDay = updateTime[0];
+            minute = updateTime[1];
+        }
+
+
+
+
+        int finalCheckedItem = checkedItem;
+        int finalHourOfDay = hourOfDay;
+        int finalMinute = minute;
+        radioGroupOne.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(RadioGroup radioGroup, int i)
+            {
+                switch (i) {
+                    case R.id.intervalRadioButton:
+
+                        spinner.setAdapter(spinnerArrayAdapter);
+                        spinner.setSelection(finalCheckedItem);
+                        spinner.setVisibility(View.VISIBLE);
+                        timePicker.setVisibility(View.GONE);
+                        string_spinnercase[0] = R.string.pref_autoUpdateIntervallOrTime_CheckStatusInterval;
+                        break;
+                    case R.id.timeRadioButton:
+                        spinner.setVisibility(View.GONE);
+                        timePicker.setVisibility(View.VISIBLE);
+                        string_spinnercase[0] = R.string.pref_autoUpdateIntervallOrTime_CheckStatusTime;
+
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
+                        {
+                            timePicker.setHour(finalHourOfDay);
+                        }
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
+                        {
+                            timePicker.setMinute(finalMinute);
+                        }
+                        break;
+                    case R.id.disableButton:
+                        spinner.setVisibility(View.GONE);
+                        timePicker.setVisibility(View.GONE);
+                        string_spinnercase[0] = R.string.pref_autoUpdateIntervallOrTime_CheckStatusNever;
+                        break;
+                    default:
+
+                }
+            }
+        });
+        radioButton_interval.setChecked(true);
+
+
+        builder.setPositiveButton(R.string.confirm_label, (dialog, which) ->
+        {
+            if (string_spinnercase[0]==R.string.pref_autoUpdateIntervallOrTime_CheckStatusInterval ) {
```

Comment:
Why does this need to be stored in a variable? Can't you just use the radio group to get the currently active view id?